assets/js/base/components/payment-methods/no-payment-methods/index.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Placeholder, Button, Notice } from 'wordpress-components';
+import { Icon, card } from '@woocommerce/icons';
+import { ADMIN_URL } from '@woocommerce/settings';
+import { useCheckoutContext } from '@woocommerce/base-context';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+/**
+ * Render content when no payment methods are found depending on context.
+ */
+const NoPaymentMethods = () => {
+	const { isEditor } = useCheckoutContext();
+
+	return isEditor ? (
+		<NoPaymentMethodsPlaceholder />
+	) : (
+		<NoPaymentMethodsNotice />
+	);
+};
+
+/**
+ * Renders a placeholder in the editor.
+ */
+const NoPaymentMethodsPlaceholder = () => {
+	return (
+		<Placeholder
+			icon={ <Icon srcElement={ card } /> }
+			label={ __( 'Payment methods', 'woo-gutenberg-products-block' ) }
+			className="wc-block-checkout__no-payment-methods"
+		>
+			<span className="wc-block-checkout__no-payment-methods-description">
+				{ __(
+					'Your store does not have any payment methods configured. Once you have configured your payment methods they will appear here.',
+					'woo-gutenberg-products-block'
+				) }
+			</span>
+			<Button
+				isDefault
+				href={ `${ ADMIN_URL }admin.php?page=wc-settings&tab=checkout` }
```

Comment:
Thinking out loud, but I wonder whether we should add `target="_blank"` here. I'm usually against forcing a link to open in a new tab, but I think this a good use-case for it. So I'm 50%-50% about adding it or leaving it as-is.